jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -75,13 +75,23 @@ public boolean isDockerInstalled() {
    *     href="https://docs.docker.com/engine/reference/commandline/load/">https://docs.docker.com/engine/reference/commandline/load/</a>
    * @return stdout from {@code docker}
    */
-  public String load(Blob imageTarballBlob) throws IOException, InterruptedException {
+  public String load(Blob imageTarballBlob) throws InterruptedException, IOException {
     // Runs 'docker load'.
     Process dockerProcess = docker("load");
 
     try (OutputStream stdin = dockerProcess.getOutputStream()) {
       imageTarballBlob.writeTo(stdin);
+
+    } catch (IOException ex) {
+      // This error usually happens when docker is not able to connect to the daemon socket.
+      try (InputStreamReader stderr =
+          new InputStreamReader(dockerProcess.getErrorStream(), StandardCharsets.UTF_8)) {
+        String error = CharStreams.toString(stderr);
```

Comment:
can 87-89 throw an IOException? If so, we might want to be careful about passing that through to the caller of `load`

Reply:
Yeah, this assumes `IOException` will be errors at the docker application level only (i.e., the docker process knows what the issue was and prints it to stderr). Technically speaking, writing to a process stdin may fail for reasons other than application level failures. For example, the docker process crashing for some unknown reason, docker command-not-found, no executable-bit set on the docker command binary, etc., and in such cases, `dockerProcess.getErrorStream()` may have nothing but the original `IOException` would better describe the situation.

Reply:
I reorganized the code a bit and passed the original `IOException` as a cause to the custom generated `IOException` so that it wouldn't be left discarded.